src/test/fakes.rs

```diff
@@ -76,15 +77,37 @@ impl<'db> FakeRelease<'db> {
         self
     }
 
+    pub(crate) fn file<M, P, D>(mut self, mimetype: M, path: P, data: D) -> Self
+        where M: Into<String>,
+              P: Into<String>,
+              D: Into<Vec<u8>>,
+        {
+        let (mimetype, path, data) = (mimetype.into(), path.into(), data.into());
+        if mimetype.contains('"') || path.contains('"') {
+            panic!("invalid mimetype or path: {}, {}", mimetype, path);
+        }
+        self.files.push((mimetype, path, data));
+
+        self
+    }
+
     pub(crate) fn create(self) -> Result<i32, Error> {
         let tempdir = tempdir::TempDir::new("docs.rs-fake")?;
 
+        let files: Vec<_> = self.files.into_iter().map(|(mimetype, path, data)| {
+            let file = tempdir.path().join(&path);
+            std::fs::write(file, data)?;
+
+            let metadata = format!("[\"{}\", \"{}\"]", mimetype, path);
+            Ok(Json::from_str(&metadata).expect("should be valid json"))
```

Comment:
Is there a reason why you're converting to string and serializing it back? This *should* work:

```rust
Ok(Json::Array(vec![Json::String(mimetype), Json::String(path)]));
```

This way you don't even need to restrict `"` in paths.

Reply:
There is not, this is better.